SkyEditor.RomEditor.Rtdx/Domain/Rtdx/Structures/Executable/MainExecutable.cs

```diff
@@ -1,125 +1,305 @@
 ﻿using NsoElfConverterDotNet;
 using SkyEditor.RomEditor.Domain.Rtdx.Constants;
 using System;
+using System.Collections;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Linq;
+using System.IO;
+using BindingFlags = System.Reflection.BindingFlags;
+#if !NETSTANDARD2_0
+using Il2CppInspector;
+using Il2CppInspector.Reflection;
+using Il2CppInspector.Model;
+#endif
 
 namespace SkyEditor.RomEditor.Domain.Rtdx.Structures.Executable
 {
-    public interface IMainExecutable
+  public interface IMainExecutable
+  {
+    IReadOnlyList<StarterFixedPokemonMap> StarterFixedPokemonMaps { get; }
+    PegasusActDatabase ActorDatabase { get; }
+    public ExecutableVersion Version { get; }
+    public byte[] Data { get; set; }
+
+    byte[] ToElf();
+    byte[] ToNso(INsoElfConverter? nsoElfConverter = null);
+
+#if !NETSTANDARD2_0
+    public AppModel IlAppModel { get; }
+    public Dictionary<string, ulong> SectionOffsets { get; }
+
+    /// <summary>Retrieve the offset of a method relative to the start of the code ELF segment.</summary>
+    ulong GetIlMethodOffset(string fullTypeName, string methodName, string[] paramTypeNames);
+
+    /// <summary>Retrieve the offset of a constructor relative to the start of the code ELF segment.</summary>
+    ulong GetIlConstructorOffset(string fullTypeName, string[] paramTypeNames);
+#endif
+  }
+
+  public class MainExecutable : IMainExecutable
+  {
+    public static MainExecutable LoadFromNso(byte[] file, byte[] il2CppMetadata, INsoElfConverter? nsoElfConverter = null)
+    {
+      nsoElfConverter ??= NsoElfConverter.Instance;
+      var data = nsoElfConverter.ConvertNsoToElf(file);
+      return new MainExecutable(data, il2CppMetadata);
+    }
+
+    public MainExecutable(byte[] elfData, byte[] il2CppMetadata)
+    {
+      if (elfData == null)
+      {
+        throw new ArgumentNullException(nameof(elfData));
+      }
+      if (elfData.Length <= (0x04BA3B0C + 0x90))
+      {
+        throw new ArgumentException("Data is not long enough to contain known sections", nameof(elfData));
+      }
+
+      this.Data = elfData ?? throw new ArgumentNullException(nameof(elfData));
+      this.Il2CppMetadata = il2CppMetadata ?? throw new ArgumentNullException(nameof(il2CppMetadata));
+
+      Init();
+    }
+
+    const int starterFixedPokemonMapOffsetOriginal = 0x04BA3B0C;
+    const int starterFixedPokemonMapOffsetUpdate = 0x04BA4DBC;
```

Comment:
I think we still can't do anything about this once since it's in a data ELF section. The offsets might be the same relative to the section though so we could try something like `SectionOffsets[".rodata"] + [some offset]`